datafusion/optimizer/src/analyzer/mod.rs

```diff
@@ -137,6 +137,19 @@ impl Analyzer {
     where
         F: FnMut(&LogicalPlan, &dyn AnalyzerRule),
     {
+        /* current test failure:
+           External error: statement is expected to fail with error:
+                   (regex) DataFusion error: check_analyzed_plan\ncaused by\nError during planning: Scalar subquery should only return one column, but found 2: t2.t2_id, t2.t2_name
+               but got error:
+                   DataFusion error: Error during planning: Scalar subquery should only return one column, but found 2: t2.t2_id, t2.t2_name
+               [SQL] SELECT t1_id, t1_name, t1_int, (select t2_id, t2_name FROM t2 WHERE t2.t2_id = t1.t1_int) FROM t1
+               at test_files/subquery.slt:436
+        */
+        // verify at the start, before the first LP analyzer pass.
+        // check_plan(&plan).map_err(|e| {
+        //     DataFusionError::Context("check_plan_before_analyzers".to_string(), Box::new(e))
+        // })?;
```

Comment:
Note this existing test failure.

TODO (if we proceed): uncomment check and debug test.